pkg/controller/perconaservermongodb/status.go

```diff
@@ -133,80 +127,33 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		}
 
 		if mongosStatus.Status != cr.Status.Mongos.Status {
-			if mongosStatus.Status == api.AppStateReady {
+			switch mongosStatus.Status {
+			case api.AppStateReady:
 				clusterCondition = api.ClusterCondition{
 					Status:             api.ConditionTrue,
-					Type:               api.ClusterMongosReady,
+					Type:               api.AppStateReady,
+					Reason:             "MongosReady",
 					LastTransitionTime: metav1.NewTime(time.Now()),
 				}
-			}
-
-			if mongosStatus.Status == api.AppStateError {
+			case api.AppStateError:
 				clusterCondition = api.ClusterCondition{
 					Status:             api.ConditionTrue,
+					Type:               api.AppStateError,
 					Message:            mongosStatus.Message,
-					Reason:             "ErrorMongos",
-					Type:               api.ClusterError,
+					Reason:             "MongosError",
 					LastTransitionTime: metav1.NewTime(time.Now()),
 				}
 			}
-			cr.Status.Conditions = append(cr.Status.Conditions, clusterCondition)
-		}
-
-		cr.Status.Mongos = &mongosStatus
-	} else {
-		cr.Status.Mongos = nil
-	}
-
-	cr.Status.State = api.AppStateInit
-	if replsetsReady == len(repls) && clusterState == clusterReady {
-		clusterCondition = api.ClusterCondition{
-			Status:             api.ConditionTrue,
-			Type:               api.ClusterReady,
-			LastTransitionTime: metav1.NewTime(time.Now()),
-		}
-		cr.Status.State = api.AppStateReady
-	} else if cr.Status.Conditions[len(cr.Status.Conditions)-1].Type != api.ClusterReady &&
-		clusterState == clusterInit {
-		clusterCondition = api.ClusterCondition{
-			Status:             api.ConditionTrue,
-			Type:               api.ClusterInit,
-			LastTransitionTime: metav1.NewTime(time.Now()),
-		}
-		cr.Status.State = api.AppStateInit
-	} else {
-		clusterCondition = api.ClusterCondition{
-			Status:             api.ConditionTrue,
-			Type:               api.ClusterError,
-			LastTransitionTime: metav1.NewTime(time.Now()),
-		}
-		cr.Status.State = api.AppStateError
-	}
-
-	if cr.Status.State != api.AppStateError &&
-		cr.Status.Mongos != nil && cr.Status.Mongos.Status != api.AppStateReady {
-		cr.Status.State = cr.Status.Mongos.Status
-	}
 
-	if len(cr.Status.Conditions) == 0 {
-		cr.Status.Conditions = append(cr.Status.Conditions, clusterCondition)
-	} else {
-		lastClusterCondition := cr.Status.Conditions[len(cr.Status.Conditions)-1]
-		switch {
-		case lastClusterCondition.Type != clusterCondition.Type:
-			cr.Status.Conditions = append(cr.Status.Conditions, clusterCondition)
-		default:
-			cr.Status.Conditions[len(cr.Status.Conditions)-1] = lastClusterCondition
+			cr.Status.AddCondition(clusterCondition)
 		}
-	}
 
-	if len(cr.Status.Conditions) > maxStatusesQuantity {
-		cr.Status.Conditions = cr.Status.Conditions[len(cr.Status.Conditions)-maxStatusesQuantity:]
+		cr.Status.Mongos = &mongosStatus
 	}
 
-	if inProgress {
-		cr.Status.State = api.AppStateInit
-	}
+	cr.Status.State = cr.Status.ClusterStatus(clusterState)
```

Comment:
Since ClusterStatus return only `Ready` or `Init` status, any `Error`s would be overshadowed by `Init` here